docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,464 @@
+However, it has not output the **Create** of Alice's `Iou` because `Iou` actions commit on Ledger 2, on which Participant Node 2 does not host Alice.
+So Alice merely *witnesses* the archival even though she is an :ref:`informee <def-informee>` of the exercise.
+The **Exercise** action is therefore marked as merely being witnessed on Participant Node 2's transaction tree stream.
+
+In general, an action is marked as **merely being witnessed** when a party is an informee of the action, but the action is not committed on a ledger on which the Participant Node hosts the party.
+Unlike **Enter** and **Leave**, such witnessed actions do not affect causality from the participant's point of view and therefore provide weaker ordering guarantees.
+Such witnessed actions show up neither in the flat transaction stream nor in the Active Contracts Service.
+
+For example, suppose that the **Create** `PaintAgree` action commits on Ledger 2 instead of Ledger 1, i.e., only the `CounterOffer` actions commit on Ledger 1.
+Then, Participant Node 2 marks the **Create** `PaintAgree` action also as merely being witnessed on the transaction tree stream.
+Accordingly, it does not report the contract as active nor can Alice use the contract in her submissions via Participant Node 2.
+
+.. _interop-causality-graph:
+
+Interoperable causality graphs
+******************************
+
+This section generalizes :ref:`causality graphs <causality-graph>` to the interoperability setting.
+
+Every active DAML contract resides on at most one DAML ledger.
+Any use of a contract must be committed on the DAML ledger where it resides.
+Initially, when the contract is created, it takes up residence on the DAML ledger on which the **Create** action is committed.
+To use contracts residing on different DAML ledgers, cross-ledger transactions are committed on several DAML ledgers.
+
+However, cross-ledger transactions incur overheads and if a contract is frequently used on a DAML ledger that is not its residence, the interoperability protocol can migrate the contract to the other DAML ledger.
+The process of the contract giving up residence on the origin DAML ledger and taking up residence on the target DAML ledger is called a **contract transfer**.
+The **Enter** and **Leave** events on the transaction stream originate from such contract transfers, as will be explained below.
+Moreover, contract transfers are synchronization points between the origin and target DAML ledgers and therefore affect the ordering guarantees.
+We therefore generalize causality graphs for interoperability.
+
+Definition »Transfer action«
+  A **transfer action** on a contract `c` is written **Transfer** `c`.
+  The **informees** of the transfer actions are the stakeholders of `c`.
+
+In the following, the term *action* refers to transaction actions (**Create**, **Exercise**, **Fetch**, and **NoSuchKey**) as well as transfer actions.
```

Comment:
Would it be worth saying that **Transfer** actions are never the consequence of another action, nor do they have consequences? Otherwise the projections section becomes confusing as one might start thinking about witnessing transfers or consequences of transfers.

Reply:
Good suggestion. Added.